### PR TITLE
ipa-server-install, ipa-server-upgrade fixes

### PR DESCRIPTION
**Pull request by MartinBasti**
ipa-server-install --setup-dns now work without BytesWarnings under python3, ipa-server-upgrade should work on IPA side but there are issues on pyldap side.

### CODE REVIEW

File: ipalib/install/sysrestore.py [hunk 1]
```diff
@@ -31,7 +31,11 @@
 
 import six
 # pylint: disable=import-error
-from six.moves.configparser import SafeConfigParser
+if six.PY3:
+    # The SafeConfigParser class has been renamed to ConfigParser in Py3
+    from configparser import ConfigParser as SafeConfigParser
+else:
+    from ConfigParser import SafeConfigParser
 # pylint: enable=import-error
```
Comment (HonzaCholasta) [inline ipalib/install/sysrestore.py]: Are the pylint pragmas still needed?

File: ipaserver/install/dnskeysyncinstance.py [hunk 2]
```diff
@@ -283,8 +283,7 @@ def __setup_replica_keys(self):
             while True:
                 # check if key with this ID exist in softHSM
                 # id is 16 Bytes long
-                key_id = "".join(chr(random.randint(0, 255))
-                                 for _ in range(0, 16))
+                key_id = bytes(random.randint(0, 255) for _ in range(0, 16))
```
Comment (HonzaCholasta) [inline ipaserver/install/dnskeysyncinstance.py]:
```
$ python2 -c 'import random; print(repr(bytes(random.randint(0, 255) for _ in range(0, 16))))'
'<generator object <genexpr> at 0x7f872517f730>'
$ python3 -c 'import random; print(repr(bytes(random.randint(0, 255) for _ in range(0, 16))))'
b'\x11\xd9\xd9\x9c\x98U\xb8w\x17\x8ar\xd6\xc6\x91\xe8\xb4'
```

File: ipalib/install/sysrestore.py (hunk 1 above)
Comment (MartinBasti) [inline ipalib/install/sysrestore.py]: Yes, still needed for pylint-3

File: ipaserver/install/dnskeysyncinstance.py (hunk 2 above)
Comment (MartinBasti) [inline ipaserver/install/dnskeysyncinstance.py]: reworked

File: ipaserver/p11helper.py [hunk 3]
```diff
@@ -1824,6 +1825,18 @@ def get_attribute(self, key_object, attr):
...
 
+def gen_key_id(key_id_len=16):
+    """
+    Generate random softhsm KEY_ID
+    :param key_id_len: this should be 16
+    :return: random softhsm KEY_ID in bytes representation
+    """
+    return struct.pack(
+        "B" * key_id_len,  # key_id must be bytes
+        *(random.randint(0, 255) for _ in range(key_id_len))
```
Comment (stlaz) [inline ipaserver/p11helper.py]: You should use `ipautil.ipa_generate_password()` here instead.

Comment (MartinBasti) [inline ipaserver/p11helper.py]: It is not password, but ID

Comment (stlaz) [inline ipaserver/p11helper.py]: Ah, nvm, saw `key` which triggered a reaction. LGTM then, need to test it.

**Comment by stlaz** [comment]
`python3-pyldap` on Fedoras 25/26 is kind of in flames: https://bugzilla.redhat.com/show_bug.cgi?id=1456673. I will see if I can get an upstream version of the package to test this properly.

**Comment by stlaz** [comment]
The failing check does not have anything in common with this PR. ACK.

File: ipaserver/install/installutils.py [hunk 4]
```diff
@@ -744,10 +749,9 @@ def read_replica_info(dir_path, rconfig):
 
     rconfig is a ReplicaConfig object
     """
-    filename = dir_path + "/realm_info"
-    fd = open(filename)
+    filename = os.path.join(dir_path, "/realm_info")
```
Comment (stlaz) [inline ipaserver/install/installutils.py]: One shall not `os.path.join()` with `/` (causes `ipa-replica-install` on DL0 to fail)

**Comment by stlaz** [comment]
ACK, I did a sanity testing, if something's wrong, some kind of our CI will hopefully tell us.

**Comment by MartinBasti** [comment]
master:

* 2e63ec42d0f879f2d129c4f81f88a1712ce86b8c py3: use ConfigParser instead of SafeConfigParser
* 6e7071d6add24e8923d705d35a362761f356d56d py3: ConfigParser: replace deprecated readfd with read
* 27f8f9f03d69276f9ee410169b76574da2461794 py3: ipaldap: encode Boolean as bytes
* d7a9e81fbd7a339dddd41a8c5ae9f29252522944 py3: softhsm key_id must be bytes
* bc9addac30d69d88f5040e194be1e32a881cfba9 py3: LDAP updates: use only bytes/raw values
* d89de4219d0e8ee33e81d6b6d1bc6c22ac9ffbaa py3: schemaupdate: fix BytesWarning
* b09a941f34507cfce682d8c5a3acf6dfe7fa624e py3: cainstance: fix BytesWarning
* c6a57d8091aeefb6067711189ee0ce11411dee57 py3: urlfetch: use "file://" prefix with filenames
* 99771ceb9ffcf21d0364bf57994716322b24551e py3: update_mod_nss_cipher_suite: ordering doesn't work with None

**Pull request closed without merging**